lib/events/encryptedevent.cpp

```diff
@@ -30,3 +31,24 @@ EncryptedEvent::EncryptedEvent(const QJsonObject& obj)
 {
     qCDebug(E2EE) << "Encrypted event from" << senderId();
 }
+
+RoomEventPtr EncryptedEvent::createDecrypted(const QString &decrypted) const
+{
+    auto eventObject = QJsonDocument::fromJson(decrypted.toUtf8()).object();
+    eventObject["event_id"] = id();
+    eventObject["sender"] = senderId();
+    eventObject["origin_server_ts"] = originTimestamp().toMSecsSinceEpoch();
+    if(contentJson().contains("m.relates_to")) {
+        auto relates = contentJson()["m.relates_to"].toObject();
+        auto content = eventObject["content"].toObject();
+        content["m.relates_to"] = relates;
+        eventObject["content"] = content;
+    }
+    if(unsignedJson().contains("redacts")) {
+        auto redacts = unsignedJson()["redacts"].toString();
+        auto unsign = eventObject["unsigned"].toObject();
+        unsign["redacts"] = redacts;
+        eventObject["unsigned"] = unsign;
+    }
+    return makeEvent<RoomMessageEvent>(eventObject);
```

Comment:
By the way, just wondering - this can be, say, `StickerEvent`, can't it?
```suggestion
    return loadEvent<RoomEvent>(eventObject);
```

Reply:
Yes, it can